### PR TITLE
Improve `README`, update `.gitignore`

### PR DESCRIPTION
**Pull request by norosa**
This patch improves the main README.rst file.

Specifically:
- Expand and improve intro description and list of features.
- Add links to key phrases to improve cross-referencing to important resources and documentation.
- Minor formatting improvements and copyediting.

### CODE REVIEW

**Comment by norosa** [comment]
this work was the initial work that ended up with me diving into the main docs so that I could add the necessary RST labels for the links I wanted to add

I have created this as a draft PR for now, because the links will not work until the rest of the changes on https://github.com/crate/crate/pull/11558/commits/801089b23412e75313d9955ac9278e0643463e4e have been merged to master (and are released)

the (and released) part is important, because changes to the CrateDB docs do now show up for the `latest` version (used in the links) until they are included in a stable release

File: README.rst [hunk 1]
```diff
@@ -22,24 +22,29 @@ About
...
+using ephemeral virtual machines (e.g., `Kubernetes`_, `AWS`_, and `Azure`_)
+with `no shared state`_. You can deploy and run CrateDB on any network—from
+personal computers to `multi-region hybrid clouds and the edge`_.
 
 
 Features
 ========
 
-- Standard SQL interface available via HTTP API and PostgreSQL wire protocol.
+- Use standard `SQL`_ via the `PostgreSQL wire protocol` or an `HTTP API`.
```
Comment (amotl) [inline README.rst:41]:
```suggestion
- Use standard `SQL`_ via the `PostgreSQL wire protocol`_ or an `HTTP API`_.
```

**Review by amotl — CHANGES_REQUESTED** [review]
Hi Naomi,

I appreciate the improvements and would like to get them merged. Can we try to make it happen after #11558 has been merged?

I also spotted two minor typos.

With kind regards,
Andreas.

File: README.rst [hunk 2]
```diff
@@ -22,24 +22,29 @@ About
...
-- Standard SQL interface available via HTTP API and PostgreSQL wire protocol.
+- Use `standard SQL`_ via the `PostgreSQL wire protocol`_ or an `HTTP API`_.
 
-- Dynamic schemas, queryable objects, geospatial features, time series data
-  support, and realtime full-text search providing functionality for handling
-  both relational and document oriented nested data structures.
+- Dynamic table schemas and queryable objects provide
+  document-oriented features in addition to the relational features of SQL.
+
+- Geospatial data types and search capability, time-series support, and realtime fulltext
```
[the remark this answers is not in the document]
Comment (amotl) [reply README.rst]: Thank you, I've updated the patch according to your suggestions.

**Pull request merged**